packages/transaction-controller/src/TransactionController.ts

```diff
@@ -285,43 +319,238 @@ const controllerName = 'TransactionController';
+ */
+export type TransactionControllerTransactionStatusUpdatedEvent = {
+  type: `${typeof controllerName}:transactionStatusUpdated`;
+  payload: [
+    {
+      transactionMeta: TransactionMeta;
+    },
+  ];
+};
+
+/**
+ * Represents the `TransactionController:transactionSubmitted` event.
+ */
+export type TransactionControllerTransactionSubmittedEvent = {
+  type: `${typeof controllerName}:transactionSubmitted`;
+  payload: [
+    {
+      transactionMeta: TransactionMeta;
+      actionId?: string;
+    },
+  ];
+};
+
+/**
+ * Represents the `TransactionController:unapprovedTransactionAdded` event.
+ */
+export type TransactionControllerUnapprovedTransactionAddedEvent = {
+  type: `${typeof controllerName}:unapprovedTransactionAdded`;
+  payload: [transactionMeta: TransactionMeta];
+};
+
+/**
+ * The internal events available to the {@link TransactionController}.
+ */
+export type TransactionControllerEvents =
+  | TransactionControllerIncomingTransactionBlockEvent
+  | TransactionControllerPostTransactionBalanceUpdatedEvent
+  | TransactionControllerSpeedupTransactionAddedEvent
+  | TransactionControllerStateChangeEvent
+  | TransactionControllerTransactionApprovedEvent
+  | TransactionControllerTransactionConfirmedEvent
+  | TransactionControllerTransactionDroppedEvent
+  | TransactionControllerTransactionFailedEvent
+  | TransactionControllerTransactionFinishedEvent
+  | TransactionControllerTransactionNewSwapApprovalEvent
+  | TransactionControllerTransactionNewSwapEvent
+  | TransactionControllerTransactionPublishingSkipped
+  | TransactionControllerTransactionRejectedEvent
+  | TransactionControllerTransactionStatusUpdatedEvent
+  | TransactionControllerTransactionSubmittedEvent
+  | TransactionControllerUnapprovedTransactionAddedEvent;
 
 /**
  * The messenger of the {@link TransactionController}.
  */
 export type TransactionControllerMessenger = RestrictedControllerMessenger<
   typeof controllerName,
-  AllowedActions,
-  AllowedEvents,
+  TransactionControllerActions | AllowedActions,
+  TransactionControllerEvents | AllowedEvents,
   AllowedActions['type'],
   AllowedEvents['type']
 >;
 
-// This interface was created before this ESLint rule was added.
-// Convert to a `type` in a future major version.
-// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
-export interface TransactionControllerEventEmitter extends EventEmitter {
-  on<T extends keyof Events>(
-    eventName: T,
-    listener: (...args: Events[T]) => void,
-  ): this;
+/**
+ * Possible states of the approve transaction step.
+ */
+export enum ApprovalState {
+  Approved = 'approved',
+  NotApproved = 'not-approved',
+  SkippedViaBeforePublishHook = 'skipped-via-before-publish-hook',
+}
 
-  emit<T extends keyof Events>(eventName: T, ...args: Events[T]): boolean;
+/**
+ * Get the default TransactionsController state.
+ *
+ * @returns The default TransactionsController state.
+ */
+function getDefaultTransactionControllerState() {
```

Comment:
What do you think about using a explicit return type `TransactionControllerState` here?

Reply:
Makes sense! Added here: [06fae6d](https://github.com/MetaMask/core/pull/3827/commits/06fae6d1165fd34dd60ad25ca86484bf900ba796)